cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1739,37 +1707,17 @@ private boolean canHaveInstances(ProgramType programType) {
   }
 
   // deletes the process metrics for a flow
-  private void deleteProcessMetricsForFlow(String application, String flow) throws IOException {
-    ServiceDiscovered discovered = discoveryServiceClient.discover(Constants.Service.METRICS);
-    Discoverable discoverable = new RandomEndpointStrategy(discovered).pick(3L, TimeUnit.SECONDS);
-
-    if (discoverable == null) {
-      LOG.error("Fail to get any metrics endpoint for deleting metrics.");
-      throw new IOException("Can't find Metrics endpoint");
-    }
-
-    LOG.debug("Deleting metrics for flow {}.{}", application, flow);
-    // TODO: use MetricStore directly to delete the metrics [CDAP-2163]
-    String url = String.format("http://%s:%d%s/metrics/system/apps/%s/flows/%s?prefixEntity=process",
-                               discoverable.getSocketAddress().getHostName(),
-                               discoverable.getSocketAddress().getPort(),
-                               Constants.Gateway.API_VERSION_3,
-                               application, flow);
-
-    long timeout = TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES);
-
-    SimpleAsyncHttpClient client = new SimpleAsyncHttpClient.Builder()
-      .setUrl(url)
-      .setRequestTimeoutInMs((int) timeout)
-      .build();
-
-    try {
-      client.delete().get(timeout, TimeUnit.MILLISECONDS);
-    } catch (Exception e) {
-      LOG.error("exception making metrics delete call", e);
-      Throwables.propagate(e);
-    } finally {
-      client.close();
-    }
+  private void deleteProcessMetricsForFlow(String namespaceId,
+                                           String application,
+                                           String flow) throws Exception {
+
+    long endTs = System.currentTimeMillis() / 1000;
+    String metricNamePrefix = "process";
+    Map<String, String> tags = Maps.newHashMap();
+    tags.put(Constants.Metrics.Tag.NAMESPACE, namespaceId);
+    tags.put(Constants.Metrics.Tag.APP, application);
+    tags.put(Constants.Metrics.Tag.FLOW, flow);
+    MetricDeleteQuery deleteQuery = new MetricDeleteQuery(0, endTs, metricNamePrefix, tags);
```

Comment:
So this is a questionable API anyway. It is to clear the queues for a flow. I guess it was introduced to allow people to clear the queues if they had bad code (or due to an update of code) so that the queues had events that cannot be processed.

It is not clear to me why we have to delete queue metrics in this case. The metrics express how many events have been processed, and that does not change by emptying the queues. Of course, if we take the difference of enqueued minus processed to determine the size of the queue... that makes sense. 

So if we delete the metrics, do we have to delete every individual metric? Or can we just delete all metrics for the flow?